src/python_testing/TC_SEPRTestBase.py

```diff
@@ -0,0 +1,190 @@
+    kMaxComponentsPerPrice = 10  # Maximum number of Component entries in PriceStruct.components list
+
+    # Test event trigger IDs
+    kEventTriggerPriceUpdate = 0x0095000000000000
+    kEventTriggerForecastUpdate = 0x0095000000000001
+
+    def check_CommodityPriceForecast(self,
+                                     cluster: Clusters.CommodityPrice,
+                                     priceForecast: list,
+                                     details: Clusters.CommodityPrice.Bitmaps.CommodityPriceDetailBitmap
+                                     = Clusters.CommodityPrice.Bitmaps.CommodityPriceDetailBitmap(0)
+                                     ):
+
+        # Verify that the DUT response contains GetDetailedForecastResponse with a
+        # list of CommodityPriceStruct entries (it may be empty) and shall have not more than 56 entries.
+        matter_asserts.assert_list(priceForecast, "PriceForecast must be a list")
+        matter_asserts.assert_list_element_type(
+            priceForecast, cluster.Structs.CommodityPriceStruct,
+            "PriceForecast list must contain CommodityPriceStruct elements",
+            allow_empty=True)
+
+        # According to spec we must not have more than 56 entries
+        asserts.assert_less_equal(len(priceForecast),
+                                  self.kMaxForecastEntries, "PriceForecast list must be less than 56 entries")
+        for item in priceForecast:
+            # The other aspects of this verification are handled by the helper
+            self.check_CommodityPriceStruct(cluster=cluster, struct=item,
+                                            details=details,
+                                            now_time_must_be_within_period=False)  # Do not check time limits for forecast
+
+    def check_CommodityPriceStruct(self,
+                                   cluster: Clusters.CommodityPrice,
+                                   struct: Clusters.CommodityPrice.Structs.CommodityPriceStruct,
+                                   details: Clusters.CommodityPrice.Bitmaps.CommodityPriceDetailBitmap = 0,
+                                   now_time_must_be_within_period: bool = True):
+        """now_time_must_be_within_period - When verifying a 'CurrentPrice' then 
+           the CurrentPrice has a single period, and so 'now' time must be within
+           the current period.
+
+           However PriceForecast also uses this same structure, which contains a
+           list of CommodityPriceStruct (going out into the future), so we
+           must no check now time is within each of the list elements.
+           """
+
+        matter_asserts.assert_valid_uint32(struct.periodStart, 'PeriodStart')
+
+        if now_time_must_be_within_period:  # Only check time limits when dealing with current price (not list of Forecast)
+            # - verify that the PeriodStart is in the past.
+            now_time_epoch_s = utc_time_in_matter_epoch() // 1000000
```

Comment:
```suggestion
            now_time_epoch_s = utc_time_in_matter_epoch() // 1_000_000
```

makes it easier to read